mod-users/src/main/resources/domain/mod-users/features/users.feature

```diff
@@ -0,0 +1,151 @@
+    * url baseUrl
+    * callonce login testUser
+    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*'  }
+    * def status = true
+    * def lastName = call random_string
+    * def firstName = call random_string
+    * def username = call random_string
+    * def email = 'abc@pqr.com'
+
+  Scenario: Create a new User with PatronGroup.
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def uuid = call uuid1
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+
+  Scenario: Search user by barcode.
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { barcode: 2222}
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { barcode: 3333}
+
+    Given path 'users?query=(barcode=2222)'
+    When method GET
+    Then status 200
+    And match response.users[0].barcode == '2222'
+
+  Scenario:  Find an active user and make that user the sponsor of another active patron
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def userId = createUserResponse.response.id
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def createProxyUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def proxyUserId = createProxyUserResponse.response.id
+
+    Given path 'proxiesfor'
+    And request {"accrueTo":"Sponsor","notificationsTo":"Sponsor","requestForSponsor":"Yes","status":"Active","proxyUserId":"#(proxyUserId)","userId":"#(userId)"}
+    When method POST
+    Then status 201
+    And match proxyUserId == response.proxyUserId
+
+  Scenario: Search user by firstname.
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { firstName: abc }
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { firstName: xyz }
+    * def responseBarcode = createUserResponse.response.firstName
+
+    Given path 'users?query=(personal.firstName=abc)'
+    When method GET
+    Then status 200
+    And match response.users[0].personal.firstName == 'abc'
```

Comment:
Similar the above, this only checks that the first user found has the expected barcode. It doesn't seem to check that no other users are found, meaning this would pass if the second user was also found. 

Maybe a check on the count of records found could avoid this.

Reply:
@marcjohnson-kint 
Yes i agree, thankyou.